### ISSUE TITLE
FFI Failure on Ubuntu 20.04.1

### ISSUE THREAD

**Issue by hellekin** [tootsuite/mastodon#14987]
### Expected behaviour

Rails console should work. `tootctl` should work. `mastodon.crontab` should run and remove cruft so that disk does not fill up.

### Actual behaviour

Ubuntu is providing `libreadline-dev` at version 8 but the FFI gem expects version 7 and fails.

```
mastodon@chat:~/live$ rails c
Traceback (most recent call last):
        42: from /home/mastodon/live/bin/rails:4:in `<main>'
        41: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `require'
        40: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        39: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `block in require'
        38: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
        37: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
        36: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        35: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
        34: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
        33: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/commands.rb:18:in `<top (required)>'
        32: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command.rb:44:in `invoke'
        31: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command.rb:70:in `find_by_namespace'
        30: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command/behavior.rb:79:in `lookup'
        29: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command/behavior.rb:79:in `each'
        28: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command/behavior.rb:80:in `block in lookup'
        27: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command/behavior.rb:80:in `each'
        26: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/command/behavior.rb:84:in `block (2 levels) in lookup'
        25: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `require'
        24: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        23: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `block in require'
        22: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
        21: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
        20: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        19: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
        18: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
        17: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/commands/console/console_command.rb:4:in `<top (required)>'
        16: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `require'
        15: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        14: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `block in require'
        13: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
        12: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
        11: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        10: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
         9: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
         8: from /usr/local/lib/ruby/2.6.0/irb/completion.rb:10:in `<top (required)>'
         7: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `require'
         6: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:257:in `load_dependency'
         5: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.4.3/lib/active_support/dependencies.rb:291:in `block in require'
         4: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
         3: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
         2: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
         1: from /home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/home/mastodon/live/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require': libreadline.so.7: cannot open shared object file: No such file or directory - /usr/local/lib/ruby/2.6.0/x86_64-linux/readline.so (LoadE
rror)
```

### Steps to reproduce the problem

Try launching the console, or run `tootctl`

### Specifications

Version v3.2.0

**Comment by hellekin** [comment]
`blurhash` requires ffi. Maybe `gem inst ffi -- --enable-system-libffi` would fix the bug...

**Comment by ThibG** [comment]
This looks like you udpated the system libs and did not rebuild native gems. Try `bundle install --force`.

**Comment by hellekin** [comment]
Great. It worked... Except I had to run `bundle` again without `--force` since ffi was rebuilt already but I got:
```
Installing e2mmap 0.1.0
Bundler::GemNotFound: Could not find e2mmap-0.1.0.gem for installation
An error occurred while installing e2mmap (0.1.0), and Bundler cannot continue.
Make sure that `gem install e2mmap -v '0.1.0' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  sidekiq-scheduler was resolved to 3.0.1, which depends on
    e2mmap
```

Thank you @ThibG!
